### PR TITLE
Refactor export-import code and resolve fixmes

### PR DESCRIPTION
**Pull request by RumovZ**
Not at all sure about 4124571af654d606020b8145828033f3fdbf7499.
This PR is supposed to address issues that should make it into 2.1.50. Let me know if there is more.

### CODE REVIEW

**Review by dae — COMMENTED** [review]
Lovely and neat! I just had two comments; everything else looks great.

**Comment by RumovZ** [comment]
Both done! 👍

File: rslib/src/storage/upgrades/mod.rs [hunk 1]
```diff
@@ -81,9 +79,10 @@ mod test {
     use super::*;
 
     #[test]
+    #[allow(clippy::assertions_on_constants)]
     fn assert_18_is_latest_schema_version() {
```
Comment (dae) [inline rslib/src/storage/upgrades/mod.rs:83]: Nice

**Comment by dae** [comment]
Thanks Rumo :-)

**Pull request merged**